packages/hardhat-core/test/internal/hardhat-network/provider/modules/eth/methods/call.ts

```diff
@@ -488,6 +494,203 @@ describe("Eth module", function () {
             assert.equal(initialBalanceAfterTx, "0xde0b6b3a7640000");
           });
         });
+
+        describe("Fee price fields", function () {
+          let deploymentBytecode: string;
+          let balanceSelector: string;
+
+          before(async function () {
+            const [_, compilerOutput] = await compileLiteral(`
+contract C {
+  function balance() public view returns (uint) {
+    return msg.sender.balance;
+  }
+}
+`);
+            const contract = compilerOutput.contracts["literal.sol"].C;
+            deploymentBytecode = `0x${contract.evm.bytecode.object}`;
+            balanceSelector = `0x${contract.evm.methodIdentifiers["balance()"]}`;
+          });
+
+          const CALLER = DEFAULT_ACCOUNTS_ADDRESSES[2];
+          let contractAddress: string;
+          let ethBalance: BN;
+
+          function deployContractAndGetEthBalance() {
+            beforeEach(async function () {
+              contractAddress = await deployContract(
+                this.provider,
+                deploymentBytecode
+              );
+
+              ethBalance = rpcQuantityToBN(
+                await this.provider.send("eth_getBalance", [CALLER])
+              );
+              assert.notEqual(ethBalance.toString(), "0");
+            });
+          }
+
+          describe("When running without EIP-1559", function () {
+            useProvider({ hardfork: "berlin" });
+
+            deployContractAndGetEthBalance();
+
+            it("Should default to gasPrice 0", async function () {
+              const balanceResult = await this.provider.send("eth_call", [
+                {
+                  from: CALLER,
+                  to: contractAddress,
+                  data: balanceSelector,
+                },
+              ]);
+
+              assert.equal(
+                rpcDataToBN(balanceResult).toString(),
+                ethBalance.toString()
+              );
+            });
+
+            it("Should use any provided gasPrice", async function () {
+              const gasLimit = 200_000;
+              const gasPrice = 2;
+
+              const balanceResult = await this.provider.send("eth_call", [
+                {
+                  from: CALLER,
+                  to: contractAddress,
+                  data: balanceSelector,
+                  gasLimit: numberToRpcQuantity(gasLimit),
+                  gasPrice: numberToRpcQuantity(gasPrice),
+                },
+              ]);
+
+              assert.isTrue(
+                rpcDataToBN(balanceResult).lt(
```

Comment:
Why is `lt` used here? In my mind, it should be either equal to that number, or maybe be between that number and the eth balance. I don't understand why it's less than that :confused:

Reply:
Fixed!